text/trace/0168-sampling-propagation.md

```diff
@@ -0,0 +1,181 @@
+`ParentBased` Sampler is used.
+
+A consistent trace sampling decision is one that can be carried out at
+any node in a trace, which supports collecting partial traces.
+OpenTelemetry specifies a built-in `TraceIDRatioBased` Sampler that
+aims to accomplish this goal but was left incomplete (see
+[TODOs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#traceidratiobased) in the specification).
+
+We propose to propagate the necessary information alongside the [W3C
+sampled flag](https://www.w3.org/TR/trace-context/#sampled-flag) using
+`tracestate` with an `otelprob` vendor tag.
+
+## Explanation
+
+Two pieces of information are needed to convey consistent head trace
+sampling probability:
+
+1. The head trace sampling probability
+2. Source of consistent sampling decisions.
+
+This proposal uses one byte of information for each of these.
+
+### Probability value
+
+To limit the cost of this extension and for statistical reasons
+documented below, we propose to limit head trace sampling probability
+to powers of two.  This limits the available head trace sampling
+probabilities to 1/2, 1/4, 1/8, and so on.  We can compactly encode
+these probabilities as small integer values using the base-2 logarithm
+of the adjusted count (i.e., inverse probability).
+
+For example, the probability value 2 corresponds with 1-in-4 sampling,
+the probability value 10 corresponds with 1-in-1024 sampling.  Using
+one byte of information we can convey sampling rates as small as 2^-255.
+
+### Random value
+
+With head trace sampling probabilities limited to powers of two, the
+amount of randomness needed per trace context is limited.  A
+consistent sampling decision is accomplished by propagating a
+geometrically distributed random variable with shape parameter `1/2`,
+requiring only two bits of randomness on average per trace.  See
+[Estimation from Partially Sampled Distributed
+Traces](https://arxiv.org/pdf/2107.07703.pdf) section 2.8 for a
+detailed explanation.
+
+Such a random variable `r` can be generated using the following
+pseudocode:
+
+```
+r := 0
+for {
+  if nextRandomBit() {
+    break // The expected value of r is 2
+  }
+  r++
+}
+```
+
+This can be computed from a stream of random bits as the number of
+leading zeros using efficient instructions on modern computer
+architectures.
+
+For example, the value 3 means there were three leading zeros and
+corresponds with being sampled at probabilities 1-in-1 through 1-in-8
+but not at probabilities 1-in-16 and smaller.  Using one byte of
+information we can convey a consistent sampling decision for sampling
+rates as small as 2^-255.
+
+### Proposed `tracestate` syntax
+
+The consistent sampling decision and head trace sampling probability
+will be propagated using four bytes of base16 content, as follows:
+
+```
+tracestate: otelprob=PPRR
+```
+
+where `PP` are two bytes of base16 probability value and `RR` are two
+bytes of base16 random value.
```

Comment:
If we limit to `2^-62` per @oertl suggestion, does it make sense still to use base 16? Base 10 is more human friendly and they both will take 2 characters to represent the max value so there is no space savings to use base 16.

Reply:
Yes, this is a good point. When @carlosalberto and I discussed this very point, we also considered that if these values were moved into the `traceparent` header, where they could really take just a byte or two, it would be more consistent to use base16 like the rest of the traceparent.

Reply:
I've added to the text that this proposal is about how we propagate, not about tracestate syntax. Whatever we decide for tracestate will dictate what we do here.